src/api/public/users/interceptors/current-user.interceptor.ts

```
@@ -0,0 +1,49 @@
+  async intercept(context: ExecutionContext, next: CallHandler) {
+    const shouldFetch = this.reflector.getAllAndOverride<boolean>(
+      FETCH_CURRENT_USER_KEY,
+      [context.getHandler(), context.getClass()],
+    );
+
+    if (shouldFetch) {
+      const req = context.switchToHttp().getRequest();
+      const userId =
+        req.headers['x-user-id'] || req.body?.userRequest?.user?.id;
+      
+
```

ESLint

[error] Delete `······⏎`

(prettier/prettier)